File: desafio2.c
#include <stdio.h>

int main(){
    int a, b;
    scanf("%d %d", &a, &b);

    int menor = a;
    if (b < menor){
        menor = b;
    }

    int i = 2;
    int mdc;
    while (i < menor){
        if (a%i == 0 && b%i == 0){
            mdc = mdc * i;
            a = a / i;
            b = b / i;
        } else {
            i++;
        }
    }

    printf("%d", mdc);

    return 0;

}

File: desafio2.py
a,b = map(int,input().split())

menor = a
if b < menor:
    menor = b

i = 2
mdc = 1
while i < menor:
    if a%i == 0 and b%i == 0:
        mdc = mdc * i
        a = a//i
        b = b//i
    else:
        i = i + 1
    

print(mdc)